open-metadata-implementation/view-services/server-author-view/server-author-view-server/src/main/java/org/odpi/openmetadata/viewservices/serverauthor/handlers/ServerAuthorViewHandler.java

```diff
@@ -0,0 +1,623 @@
+
+        configuredPlatforms = new HashMap<>();
+        configuredServerInstances = new HashMap<>();
+
+        if (resourceEndpoints != null && !resourceEndpoints.isEmpty()) {
+
+            resourceEndpoints.forEach(res -> {
+
+                String resCategory = res.getResourceCategory();
+                ResourceEndpoint rep = new ResourceEndpoint(res);
+
+                String resName = null;
+
+                switch (resCategory) {
+                    case "Platform":
+                        resName = res.getPlatformName();
+                        configuredPlatforms.put(resName, rep);
+                        break;
+                    // At this time there is no need to pick up servers from the config for this view service.
+                    default:
+                        // Unsupported category is ignored
+                        break;
+
+                }
+            });
+        }
+    }
+
+    /**
+     * getResourceEndpoints - returns a list of the configured resource endpoints. Does not include discovered resource endpoints.
+     *
+     * @param userId     userId under which the request is performed
+     * @param methodName The name of the method being invoked
+     *                   This method will return the resource endpoints that have been configured for the view service
+     */
+    public Map<String, List<ResourceEndpoint>> getResourceEndpoints(String userId, String methodName) {
+
+        Map<String, List<ResourceEndpoint>> returnMap = new HashMap<>();
+
+        List<ResourceEndpoint> platformList = null;
+        List<ResourceEndpoint> serverList = null;
+
+        if (!configuredPlatforms.isEmpty()) {
+            platformList = new ArrayList<>();
+            platformList.addAll(configuredPlatforms.values());
+        }
+
+        if (!configuredServerInstances.isEmpty()) {
+            serverList = new ArrayList<>();
+            serverList.addAll(configuredServerInstances.values());
+        }
+
+        returnMap.put("platformList", platformList);
+
+        return returnMap;
+    }
+
+
+    /**
+     * resolvePlatformRootURL
+     * <p>
+     * This method will look up the configured root URL for the named platform.
+     *
+     * @param platformName
+     * @return resolved platform URL Root
+     * @throws org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException
+     */
+    private String resolvePlatformRootURL(String platformName, String methodName) throws InvalidParameterException {
+        String platformRootURL = null;
+
+        if (platformName != null) {
+            ResourceEndpoint resource = configuredPlatforms.get(platformName);
+            if (resource != null) {
+                platformRootURL = resource.getResourceRootURL();
+            }
+        }
+        if (platformName == null || platformRootURL == null) {
+            throw new InvalidParameterException(ServerAuthorViewErrorCode.VIEW_SERVICE_NULL_PLATFORM_NAME.getMessageDefinition(),
```

Comment:
I have altered this in the latest Rex Handler - it now maps the exception here into the view service specific exception type. You don't have to do that but I thought it helped.